tensorflow/compiler/mlir/tensorflow/translate/mlir_roundtrip_flags.h

```diff
@@ -38,6 +38,12 @@ struct ArrayInfo {
   TensorShapeProto shape;
 };
 
+struct ArrayInfo : public ArrayInfoBase {
```

Comment:
Why did you add a subclass? I don't see the base case used anywhere.

Reply:
So initially I just wanted to just add a 'std::vector<ArrayInfo> subtypes' to the ArrayInfo class, however since the subtypes are not going to have subtypes themselves, I decided to factor out shape and imported_dtype into a base class and alias SubTypeInfo to be the base class, and have ArrayInfo inherit from base and have a std::vector<SubTypeInfo>. This was the cleanest solution I could think of

Reply:
Gotcha.